### ISSUE TITLE
Multiple sections, replace all items in section and notify observers

### ISSUE THREAD

**Issue by mtaweel** [ReactiveKit/Bond#338]
Hello,
I'm trying to use `MutableObservable2DArray` to support multiple sectioned table. Is there a way to replace all items in a certain section and perform diff and notify observers? I can't find it anywhere in the `MutableObservable2DArray` available methods

Thx

**Comment by srdanrasic** [comment]
Unfortunately this is not supported out of the box at the moment. It's good suggestion for an improvement though.

**Comment by mtaweel** [comment]
After taking a look on how you did it with `MutableObservableArray`, I've added the following to `MutableObservable2DArray` to support replace with diff, should I submit a pull request with this change? what do you think?

```
extension MutableObservable2DArray {
    
    public func replaceSection(at index: Int, with section: Observable2DArraySection<SectionMetadata, Item>)  {
        lock.lock(); defer { lock.unlock() }
        sections[index] = section
        subject.next(Observable2DArrayEvent(change: .updateSections([index]), source: self))
    }
}

extension MutableObservable2DArray where Item: Equatable {
    
    public func replaceSection(at index: Int, with section: Observable2DArraySection<SectionMetadata, Item>, performDiff: Bool) {
        if performDiff {
            lock.lock()
            let diff = Array.diff(sections[index].items, section.items)
            
            var deletes: [Int] = []
            var inserts: [Int] = []
            deletes.reserveCapacity(diff.count)
            inserts.reserveCapacity(diff.count)
            
            for diffStep in diff {
                switch diffStep {
                case .insert(_, let index):
                    inserts.append(index)
                case .delete(_, let index):
                    deletes.append(index)
                }
            }
            let deletesIndexPaths = deletes.map { IndexPath(item: $0, section: index) }
            let insertsIndexPaths = inserts.map { IndexPath(item: $0, section: index) }
            
            subject.next(Observable2DArrayEvent(change: .beginBatchEditing, source: self))
            sections[index].metadata = section.metadata
            sections[index].items = section.items
            subject.next(Observable2DArrayEvent(change: .deleteItems(deletesIndexPaths), source: self))
            subject.next(Observable2DArrayEvent(change: .insertItems(insertsIndexPaths), source: self))
            subject.next(Observable2DArrayEvent(change: .endBatchEditing, source: self))
            lock.unlock()
        } else {
            replaceSection(at: index, with: section)
        }
    }
    
    public func replaceSection(at index: Int, with items: [Item], performDiff: Bool) {
        replaceSection(at: index, with: Observable2DArraySection<SectionMetadata, Item>(metadata: sections[index].metadata, items: items), performDiff: performDiff)
    }
}

```

**Comment by srdanrasic** [comment]
That looks nice! If you can do a PR I'd be happy to merge it.

**Comment by mtaweel** [comment]
Done! https://github.com/ReactiveKit/Bond/pull/339

**Comment by prefect42** [comment]
Haha. Okay. I did not see this when writing the other issue a minute ago 🎉

**Comment by prefect42** [comment]
Please merge this!

**Comment by srdanrasic** [comment]
Released in v5.3.0.
